heat/core/operations.py

```diff
@@ -509,6 +515,494 @@ def sum(x, axis=None, out=None):
     return __reduce_op(x, torch.sum, MPI.SUM, axis, out)
 
 
+def merge_means(mu1, n1, mu2, n2):
+    """
+    Function to merge two means by pairwise update
+
+    Parameters
+    ----------
+    mu1 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n1 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu1
+    mu2 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n2 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu2
+
+    Returns
+    -------
+    mean of combined set
+    number of elements in the combined set
+
+    References
+    ----------
+    [1] J. Bennett, R. Grout, P. Pebay, D. Roe, D. Thompson, Numerically stable, single-pass, parallel statistics algorithms,
+        IEEE International Conference on Cluster Computing and Workshops, 2009, Oct 2009, New Orleans, LA, USA.
+    """
+    delta = mu2.item() - mu1.item()
+    n1 = n1.item()
+    n2 = n2.item()
+    return mu1 + n2 * (delta / (n1 + n2)), n1 + n2
+
+
+def mean(x, axis=None):
+    """
+    Calculates and returns the mean of a tensor.
+    If a axis is given, the mean will be taken in that direction.
+
+    Parameters
+    ----------
+    x : ht.tensor
+        Values for which the mean is calculated for
+    axis : None, Int, iterable
+           axis which the mean is taken in.
+           Default: None -> mean of all data calculated
+
+    Returns
+    -------
+    ht.tensor containing the mean/s, if split, then split in the same direction as x.
```

Comment:
A number of example could be useful here. Maybe you can just use/adapt the ones on the numpy or torch mean() page.